### ISSUE TITLE
Support multiple packages in a repo via plugin

### ISSUE THREAD

**Issue by NilsJacobsen** [inlang/inlang#985]
According to this PR @Maskbook -> https://github.com/DimensionDev/Maskbook/pull/9889

Maskbook has multiple apps with i18next locales folders in their repo. 
So right in the IDE-extension there works only one instance.

#### Ideas:
I guess there is no super easy quick fix for this.
- Using ide extension settings
- Using the config settings
- using a root config file and multiple sub configs
- ...

#### Why should we care:
- The bigger projects get the more likey it is that people are working isolated on only one app with their own stack and i18n solution, I guess also corporates build like that. 

Any ideas on that @inlang/team

**Comment by NilsJacobsen** [comment]
May think of it like this:
How can it be achieved hands-on to ship tomorrow?
How should it be done in half a year?

**Comment by NilsJacobsen** [comment]
Maybe as a quick fix, we could use a plugin setting to specify a path to the config. That way they could change the path while working on a specific app...

**Comment by NilsJacobsen** [comment]
Hmm hacking it together would mean creating a hack plugin that could read multible files. Then I guess we need another prefix because we don't want duplicate keys right?

**Comment by felixhaeberle** [comment]
👎 Against hacking. Monorepo support should work across every app and therefore an initiative inlang core should lead, or, should be enabled by the specific app in an general exposed API like the `--config` for the CLI or a potential `configPath` in the `ideExtension` setting. 

TLDR: This does not mean we can't establish monorepo support in a specific app, for example:
- we already have monorepo support in the CLI where we gave the possibility to read from a specific config file path.

**This should probably be a VS Code Extension setting.**

**Comment by samuelstroschein** [comment]
@NilsJacobsen 

> Then I guess we need another prefix because we don't want duplicate keys right?

Correct. 

An intuitive API could be 

```ts
i18nextPlugin({
  pathPattern: {
    "project-a": "./package-a/translations/{language}.json",
    "project-b": "./package-b/locales/{language}.json"
  }
})
```

project-a and project-b are the prefixes. proper monorepo support would ofc be better

**Comment by NilsJacobsen** [comment]
Monorepo or personal fork?

**Comment by felixhaeberle** [comment]
Oh, I guess there are two issues here? 
1. Which config to choose 
2. which resources to choose

**Comment by samuelstroschein** [comment]
> @felixhaeberle  This is already fixed in the CLI btw. 😄

But not in the editor, vscode extension, every other inlang app :)

**Comment by samuelstroschein** [comment]
Yoo wait, this API https://github.com/inlang/inlang/issues/985#issuecomment-1602541992 likely also solves a lot of "namespacing" problems in general. 

```ts

i18nextPlugin({
  pathPattern: {
    // we don't need implicit namespacing anymore
    "project-a.common": "./package-a/translations/{language}/common.json",
    "project-a.mobile": "./package-a/translations/{language}/mobile.json",
    "project-b": "./package-b/locales/{language}.json"
  }
})
```

@NilsJacobsen seems like a good idea in general?

**Comment by NilsJacobsen** [comment]
Uhhh that sounds interesting. Gonna explore that further.

**Comment by NiklasBuchfink** [comment]
## Support Monorepos

The more I think through Option 2 it makes sense. Just extend the `inlang.config.js` with workspace functionality and bring it step by step to all apps. We should keep the colocation of the `inlang.config.js` with the related "app".

Sadly there is no way to just skim through the repo and find all `inlang.config.js` files before partial cloning of the files. Maybe a `project-lisa` thing to request locations from the provider before the execution of a partial clone. Something like this:

```js
// get config locations
const workspaces = await lisa.located("inlang.config.js")

console.log(workspaces)
// [ "./package-a/translations/{language}/common.json", "./package-a/translations/{language}/mobile.json"]

// clone workspace according to user selection
```

## Namespaces

I actually like the `*` structure because it makes it easy to split up the translations for specific parts of the app, without adding for example 6 static paths. The related issues with that are just because it's not dissolved correctly.

**Comment by samuelstroschein** [comment]
@NilsJacobsen @felixhaeberle and I had a call. Summary: 

https://github.com/DimensionDev/Maskbook/pull/9889 required a lot of effort to make working. If they don't have super high interest, probably not worth it. 

- workspace support is required in the long run 
- namespace support is also required in the long run 
- short term, focus on repos that don't require workspace support

Furthermore, the chain to introduce functionality in inlang is as follows: 

1. Lowest effort: Plugins [no app, nothing in core needs to change]
2. Medium effort: Apps [no plugins need change but every app]
3. High effort: Core [plugins and apps need to change] 

Please keep the chain in mind when you think about introducing features. Features introduced in plugins can "climb up the ladder" over time.

**Comment by samuelstroschein** [comment]
@NilsJacobsen i default assigned you to this issue, ensuring that this issue will be completed/closed as not a prio

**Comment by jonator** [comment]
https://code.visualstudio.com/docs/editor/workspaces

Leaving this here, as it's my preferred way of working with mono repos in vscode (it fixes a lot of my extensions, like jest for example).

**Comment by NilsJacobsen** [comment]
Update: https://github.com/inlang/inlang/pull/1015 
With this PR, you can set up one config file that points to different parts of the monorepo and prefixes the keys with custom prefixes. This way, you can use multiple resource folders. 

https://github.com/DimensionDev/Maskbook/pull/9889 -> For Maskbook, creating a custom parser to enable their unique typesafe setup is a feasible option for them.
